coderd/database/queries/notificationsinbox.sql

```diff
@@ -0,0 +1,38 @@
+-- name: FetchUnreadInboxNotificationsByUserID :many
+SELECT * FROM inbox_notifications WHERE user_id = $1 AND read_at IS NULL ORDER BY created_at DESC;
+
+-- name: FetchInboxNotificationsByUserID :many
+SELECT * FROM inbox_notifications WHERE user_id = $1 ORDER BY created_at DESC;
+
+-- name: FetchInboxNotificationsByUserIDFilteredByTemplatesAndTargets :many
+SELECT * FROM inbox_notifications WHERE user_id = @user_id AND template_id = ANY(@templates::UUID[]) AND targets @> COALESCE(@targets, ARRAY[]::UUID[]) ORDER BY created_at DESC;
```

Comment:
I had to look this up.

https://www.postgresql.org/docs/current/functions-array.html

`@>` implies that _every_ element in `COALESCE(@targets, ARRAY[]::UUID[])` must appear in `targets`. Seems reasonable since the filtering will be ANDs not ORs.

In any case, it would be useful to document the behaviour that callers can expect from this query, and why you chose this specific operator.